internal/pipeline/pipeline.go

```diff
@@ -228,6 +248,68 @@ func (f *Function) Validate() error {
 	return nil
 }
 
+// runner returns a fnRunner from the image and configs of
+// this function.
+func (f *Function) runner() (kio.Filter, error) {
+	config, err := f.config()
+	if err != nil {
+		return nil, err
+	}
+	return &fnRunner{
+		fn: &runtime.ContainerFn{
+			Image: f.Image,
+		},
+		fnConfig: config,
+	}, nil
+}
+
+func (f *Function) config() (*yaml.RNode, error) {
+	var node *yaml.RNode
+	switch {
+	case f.ConfigPath != "":
+		file, err := os.Open(f.ConfigPath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to open config file path %s: %w", f.ConfigPath, err)
+		}
+		b, err := ioutil.ReadAll(file)
+		if err != nil {
+			return nil, fmt.Errorf("failed to read content from config file: %w", err)
+		}
+		node, err = yaml.Parse(string(b))
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse config %s: %w", string(b), err)
+		}
+		// directly use the config from file
+		return node, nil
+	case !isNodeZero(&f.Config):
+		// directly use the inline config
+		return yaml.NewRNode(&f.Config), nil
+	case len(f.ConfigMap) != 0:
+		node = yaml.NewMapRNode(&f.ConfigMap)
```

Comment:
Logic after the switch should only be done for this case:

return wrapConfigMap(node)

Reply:
done